### PR TITLE
Feature: High level aggregated CSV download for sitemaps.

### LINKED ISSUE #809: Please provide an aggregated CSV data file in downloaded PSAT CLI Report
## Feature Request

**Description:**
[Provide a clear and concise description of the feature you are requesting.]
After generating an aggregated report using PSAT CLI, the option to download the report should also provide and aggregated .csv file containing the top level cookie count data for each URL in the aggregate report

**Motivation:**
[Explain the motivation or problem that this feature will address. What value will it bring to the users or the project?]
When downloading a PSAT report, an copy of the Aggregated HTML and URL specific .csv's are provided in the zip folder. This is useful, but to compare the cookie data across all the URLs the user must open each CSV file individually. Having a way to view all of the high level URL data in a single view would be helpful for analysis

**User Story:**
As a frequent PSAT CLI user, I want a complete high level view so that I can compare the 3P cookies blockage frequencies across a number of sites

**Acceptance Criteria:**
- [ ] Downloaded zip folder contains report.html as well as a report.csv containing the Total, 1P and 3P cookie counts in a single top level csv file 

**Additional Information:**
[Include any additional information or context that may be helpful, such as related resources or examples.]

### PR DESCRIPTION
**Pull request by amovar18**
## Description
This PR aims to allow users to download aggregated CSV data when analysing a sitemap.

## Relevant Technical Choices
- A new function to generate aggregated CSV has been introduced to calculate the aggregated data of the sitemap.

## Testing Instructions
- Clone this branch
- In the terminal run `npm run build:cli:dashboard`.
- Then run analysis on a sitemap.
- After analysis is done, open the report and download the report from sitemap landing page.
- You should see a `report.csv` in the zip file, which will contain the data of the landing page.

## Screenshot/Screencast

---

## Checklist

- [x] I have thoroughly tested this code to the best of my abilities.
- [x] I have reviewed the code myself before requesting a review.
- [x] This code is covered by unit tests to verify that it works as intended.
- [x] The QA of this PR is done by a member of the QA team (to be checked by QA).

Fixes #809

### CODE REVIEW

File: packages/cli/src/utils/saveReports.ts [hunk 1]
```diff
@@ -54,6 +58,10 @@ const saveReports = async (
       'report.html',
       sitemapUrl
     );
+    console.log(extractReportData(result).landingPageCookies);
```
Comment (mohdsayed) [inline packages/cli/src/utils/saveReports.ts:61]: Want to keep it?

File: packages/common/src/utils/extractCookies.ts [hunk 2]
```diff
@@ -14,15 +14,11 @@
...
-import {
-  calculateEffectiveExpiryDate,
-  type CompleteJson,
-  type CookieFrameStorageType,
-} from '@google-psat/common';
-
 /**
  * Internal dependencies
  */
+import { CompleteJson, CookieFrameStorageType } from '../cookies.types';
```
Comment (mohdsayed) [inline packages/common/src/utils/extractCookies.ts:20]: Please use `type`

File: packages/common/src/utils/generateReports/generateRootSummaryDataCSV.ts [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    [I18n.getMessage('uncategorizedCookiesWithIssues')]:
+      uncategorisedCookiesWithIssues,
+  };
+
+  const CSVString = Object.entries(summary).reduce(
+    (acc, [key, value]) => (acc += `${key}, ${value}\r\n`),
+    ''
+  );
+
+  return CSVString;
```
Comment (mohdsayed) [inline packages/common/src/utils/generateReports/generateRootSummaryDataCSV.ts:132]:
```suggestion
 return Object.entries(summary)
    .map(([key, value]) => `${key}, ${value}`)
    .join('\r\n');
```

Will that return the same result?

File: packages/extension/src/utils/tests/generateReportObject.ts [hunk 4]
```diff
@@ -37,9 +37,12 @@ describe('generateReport', () => {
         message: 'Exempted Cookies',
       },
     });
+
     globalThis.fetch = jest.fn().mockResolvedValue({
       json: jest.fn().mockResolvedValue({}),
     });
+
+    //@ts-ignore
```
Comment (mohdsayed) [inline packages/extension/src/utils/tests/generateReportObject.ts:45]: Let's keep adding reasons.

File: packages/common/src/utils/generateReports/generateRootSummaryDataCSV.ts (hunk 3 above)
Comment (amovar18) [reply packages/common/src/utils/generateReports/generateRootSummaryDataCSV.ts:132]: `.join` would not add `\r\n` at the end of the string. We can modify and add it in the string before returning.

File: packages/common/src/utils/extractCookies.ts (hunk 2 above)
Comment (amovar18) [reply packages/common/src/utils/extractCookies.ts]: Fixed in [d2aca92](https://github.com/GoogleChromeLabs/ps-analysis-tool/pull/814/commits/d2aca929c6b25a68d828a6bd79e4e422e6b91e5f)

File: packages/extension/src/utils/tests/generateReportObject.ts (hunk 4 above)
Comment (amovar18) [reply packages/extension/src/utils/tests/generateReportObject.ts]: Fixed in [d2aca92](https://github.com/GoogleChromeLabs/ps-analysis-tool/pull/814/commits/d2aca929c6b25a68d828a6bd79e4e422e6b91e5f)

**Review by mohdsayed — APPROVED** [review]
Looks good.

**Pull request merged**